src/main/java/de/tum/in/www1/artemis/web/rest/hestia/CodeHintResource.java

```diff
@@ -48,12 +49,27 @@ public CodeHintResource(AuthorizationCheckService authCheckService, ProgrammingE
         this.codeHintService = codeHintService;
     }
 
+    /**
+     * GET programming-exercises/{exerciseId}/code-hints: Retrieve all code hints for a programming exercise.
+     * @param exerciseId of the exercise
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the code hints for the exercise
+     */
+    @GetMapping("programming-exercises/{exerciseId}/code-hints")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<Set<CodeHint>> getAllCodeHints(@PathVariable Long exerciseId) {
```

Comment:
Any particular reason why this endpoint is only accessible for Editors? `ExerciseHintResource::getExerciseHintsForExercise` is available for Tutors, so it would make sense to keep this consistent.

Reply:
Fair point. The `Manage hints` page (and related routes) in the client are currently only accessible to editors. Because the hints are not loaded for the PS anymore, the endpoints cannot be accessed by tutors within the UI at all.

So either we change the authorities, or we adapt the endpoint authorization. I would prefer the first option.